v2/component/launcher_v2.go

```diff
@@ -0,0 +1,149 @@
+package component
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/kubeflow/pipelines/api/v2alpha1/go/pipelinespec"
+	"github.com/kubeflow/pipelines/v2/metadata"
+	"github.com/kubeflow/pipelines/v2/objectstore"
+	pb "github.com/kubeflow/pipelines/v2/third_party/ml_metadata"
+	"google.golang.org/protobuf/encoding/protojson"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type LauncherV2Options struct {
+	Namespace,
+	PodName,
+	PodUID,
+	PipelineRoot,
+	MLMDServerAddress,
+	MLMDServerPort string
+}
+
+type LauncherV2 struct {
+	executionID   int64
+	executorInput *pipelinespec.ExecutorInput
+	command       string
+	args          []string
+	options       LauncherV2Options
+
+	// clients
+	metadataClient *metadata.Client
+	k8sClient      *kubernetes.Clientset
+}
+
+func NewLauncherV2(executionID int64, executorInputJSON string, cmdArgs []string, opts *LauncherV2Options) (l *LauncherV2, err error) {
+	defer func() {
+		if err != nil {
+			err = fmt.Errorf("failed to create component launcher v2: %w", err)
+		}
+	}()
+	if executionID == 0 {
+		return nil, fmt.Errorf("must specify execution ID")
+	}
+	executorInput := &pipelinespec.ExecutorInput{}
+	err = protojson.Unmarshal([]byte(executorInputJSON), executorInput)
+	if err != nil {
+		return nil, err
```

Comment:
Do we need to add prefix in err msg like "failed to unmarshall xxx"?

Reply:
Good point, the existing error message cannot tell it's for executor input.
Updated